asm-lsp/bin/main.rs

```diff
@@ -189,9 +188,31 @@ pub fn main() -> Result<()> {
 
     let arm_instructions = if config.instruction_sets.arm.unwrap_or(false) {
         let start = std::time::Instant::now();
-        let arm_instrs = include_bytes!("../../docs_store/opcodes/serialized/arm");
-        // NOTE: No need to filter these instructions by assembler like we do for
-        // x86/x86_64, as our ARM docs don't contain any assembler-specific information (yet)
+        let arm_instrs = include_bytes!("../serialized/opcodes/arm");
+        // NOTE: Actually, the arm file are all arm64 so we needed to get
+        // the arm32 versions then do the below
+        // NOTE: No need to filter these instructions by assembler
+        // like we do for x86/x86_64, as our ARM docs don't contain any
+        // assembler-specific information (yet)
+        let instrs = bincode::deserialize::<Vec<Instruction>>(arm_instrs)?;
+        info!(
+            "arm instruction set loaded in {}ms",
+            start.elapsed().as_millis()
+        );
+        instrs
+    } else {
+        Vec::new()
+    };
+
+    let arm64_instructions = if config.instruction_sets.arm64.unwrap_or(false) {
+        let start = std::time::Instant::now();
+        // TODO: change to arm64 after arm32 has been added
+        let arm_instrs = include_bytes!("../serialized/opcodes/arm");
```

Comment:
Oh I see, that's some work for a future PR.

Reply:
Currently the file in the opcodes directory are actually arm64 files, I plan to add the arm32 file later but that isn't a priority for me since I'm currently on focused on arm64 and riscv64. But when I get around to it, I plan to parse and write relevant parts of all the arm64 and arm32 mnemonics into a single arm32.xml and arm64.xml because I realized in the parsing code that we actually only need a few sections of each of the whole opcode files